packages/astro/src/runtime/server/endpoint.ts

```diff
@@ -1,28 +1,55 @@
 import type { APIContext, EndpointHandler, Params } from '../../@types/astro';
+import { type LogOptions, warn, info } from '../../core/logger/core.js';
 
-function getHandlerFromModule(mod: EndpointHandler, method: string) {
+function getHandlerFromModule(mod: EndpointHandler, method: string, logging: LogOptions) {
+	const lowerCaseMethod = method.toLowerCase();
+
+	// NOTE: remove in Astro 4.0
+	if (mod[lowerCaseMethod]) {
+		warn(
+			logging,
+			'astro',
+			`Lower case endpoint names are deprecated and will not be supported in Astro 4.0. Rename the endpoint ${lowerCaseMethod} to ${method}.`
```

Comment:
Does this logging tell users which file/path they need to update?

Reply:
Unfortunately we don't have this information in that point of the code :(